From e9e90634c7da6439d377e6028b473c80dc20f4f0 Mon Sep 17 00:00:00 2001
From: Jerry-wiwynn <Jerry_C_Chen@wiwynn.com>
Date: Wed, 15 May 2024 14:06:23 +0800
Subject: [PATCH 502/504] mctpd: fix mctpd crash issue

Add null check in find_peer_by_association() to prevent access violation
---
 src/mctpd.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/mctpd.c b/src/mctpd.c
index f11d716..b4e7cae 100644
--- a/src/mctpd.c
+++ b/src/mctpd.c
@@ -436,7 +436,7 @@ static struct peer * find_peer_by_association(struct ctx *ctx, const char *objec
 {
 	for (size_t i = 0; i < ctx->num_peers; i++) {
 		struct peer *peer = ctx->peers[i];
-		if (peer->state != REMOTE)
+		if (peer->state != REMOTE || peer->association == NULL)
 			continue;
 		if (strcmp(peer->association->object_path, object_path) == 0)
 			return peer;
@@ -5051,7 +5051,8 @@ static int on_interface_removed(sd_bus_message *m, void *userdata, sd_bus_error
 
 	for (peer = find_peer_by_association(ctx, temp); peer;
 		 peer = find_peer_by_association(ctx, temp)) {
-		remove_peer(peer);
+		if (peer)
+			remove_peer(peer);
 	}
 
 	return rc;
-- 
2.52.0

